helix-core/src/test/java/org/apache/helix/monitoring/mbeans/TestDropResourceMetricsReset.java

```diff
@@ -148,6 +148,10 @@ public void testDropWithNoCurrentState() throws Exception {
     // Drop the resource
     setupTool.dropResourceFromCluster(clusterName, RESOURCE_NAME);
 
+    // TEMP WORKAROUND, trigger a CurrentStateChange to remove the resource monitor
+    // https://github.com/apache/helix/issues/1980
+    participant.syncStart();
```

Comment:
could we give a little more context? (e.g. adding a liveinstance has an effect of creating a CurrentStageChange event, **which in turn triggers ExternalViewStage that includes the metric cleanup logic**)

also, should we stop this participant when we clean up (line 159 and beyond)? otherwise, it will leave a dangling thread.

Reply:
Good point, updated. Thanks.